// Copyright 2019 the gRPC authors.

syntax = "proto3";

package hash_name;

message HashNameRequest {

  string desired_name = 1;

  int32 ideal_hamming_distance = 2;

  int32 interesting_hamming_distance = 3;
}

message HashNameResponse {

  string secret = 1;

  string hashed_name = 2;

  int32 hamming_distance = 3;
}

service HashFinder {

  rpc Find (HashNameRequest) returns (HashNameResponse) {}

  rpc FindRange (HashNameRequest) returns (stream HashNameResponse) {}
}
